### PR TITLE
fix(yaml): bound configuration resource usage

### PR DESCRIPTION
**Pull request by abechan1**
### Motivation

- Prevent untrusted or malformed YAML from exhausting CPU/memory by bounding input size, total parsed nodes, scalar length, and collection element counts.
- Fail early with a path-aware error when limits are exceeded so callers can surface actionable messages instead of crashing or hanging.
- Apply the same protections to both file-based and in-memory text inputs and keep valid wip/Compose files working at the boundary.

### Description

- Introduce runtime limits and constants: `MaxInputSize`, `MaxTotalNodes`, `MaxScalarLength`, and `MaxCollectionElements` in `src/Wip.Core/Yaml/YamlLoader.cs` and wire them into the loader.
- Check `FileInfo.Length` before opening streams in `LoadFile`, and enforce a streaming byte limit via a `LimitedReadStream` while reading; apply `MaxInputSize` to `LoadText` as well.
- Add a `LoadState` node counter that is passed to `ReadNode` and incremented per parsed node, and validate `MaxTotalNodes`; validate scalar length on scalar events and enforce per-collection element limits in `ReadSequence` and `ReadMapping`.
- Emit a path-aware `ConfigException` via a new helper `LimitExceeded` when any limit is exceeded, add security-focused unit tests in `tests/Wip.Tests/YamlLoaderSecurityTests.cs`, and bump the package version to `2.3.2`.

### Testing

- Ran `git diff --check` to validate the patch formatting and it passed.
- Added unit tests in `tests/Wip.Tests/YamlLoaderSecurityTests.cs` covering oversized scalar, excessive sequence/mapping elements, total node count, oversized text/file inputs, boundary acceptance of normal `wip.yml`, and alias-enabled Compose file at the input boundary; these tests were added but not executed in this environment because `dotnet` is not available here (`dotnet test --no-restore` could not be run).
- Attempted to use the local PR helper tooling to record the PR metadata but the helper could not complete due to environment/network constraints; this does not affect the code changes or the added tests.

------
[Codex Task](https://chatgpt.com/codex/cloud/tasks/task_e_6a97501c85188322832f9ddc8576d169)

## Summary by CodeRabbit

* **New Features**
  * Added safeguards for YAML input size, scalar length, collection size, total node count, and merge expansion.
  * Oversized or overly complex YAML files and text inputs are now rejected with clear limit errors.
  * Valid files at supported boundaries, including files using aliases, continue to load successfully.

* **Tests**
  * Added coverage for YAML security limits and boundary conditions.

* **Chores**
  * Updated the project version to 2.3.2.

### CODE REVIEW

File: src/Wip.Core/Yaml/YamlLoader.cs [hunk 1]
```diff
@@ -26,15 +26,32 @@ public static class YamlLoader
...
+        using var reader = new StreamReader(limitedStream);
         return Load(reader, allowAliases, path);
     }
 
     public static object? LoadText(string text, bool allowAliases, string path = "<text>")
     {
+        if (text.Length > MaxInputSize)
+        {
+            throw LimitExceeded(path, "input length", MaxInputSize);
+        }
```
[the remark this answers is not in the document]
Comment (abechan1) [reply src/Wip.Core/Yaml/YamlLoader.cs]: Fixed in f31b079. `LoadText` now measures `Encoding.UTF8.GetByteCount(text)` instead of `text.Length`, so in-memory text and file input are held to the same `MaxInputSize` bound in bytes. The message wording changed from "input length" to "input size" to match, and `RejectsTextThatOnlyExceedsTheLimitInUtf8Bytes` covers a document that is under the limit in characters but over it in UTF-8 bytes.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by abechan1** [comment]
Addressed the review feedback in f31b079.

**Merge expansion (CodeRabbit's merge risk).** Confirmed: the node limits do not bound merge work. An alias makes a merge source shareable, so `<<: [*big, *big, ...]` costs one node per copy but copies the whole source each time, and repeating that across mappings multiplies it again — a sub-1 MiB file could ask for on the order of a billion dictionary writes while staying inside `MaxTotalNodes`. `MergeInto` now charges each copied entry against a per-document `MaxMergeEntries` budget (1,000,000) and fails with the same path-aware limit error. `RejectsMergeExpansionOverTheLimit` and `AcceptsMergeExpansionAtTheLimit` cover the boundary.

**UTF-8 vs UTF-16 input bound (Copilot).** `LoadText` now measures UTF-8 bytes, matching the file path; replied on that thread.

**Docstrings.** Added XML docs to the limit constants and the loader's new internal helpers.

Note: `dotnet` is not available in this environment and the SDK download is blocked by the network policy, so the new tests were not executed locally — CI is the first run.

---
_Generated by [Claude Code](https://claude.ai/code)_
